src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawWriter.cs

```diff
@@ -218,6 +218,15 @@ public override void WriteNode(System.Xml.XPath.XPathNavigator navigator, bool d
             throw new InvalidOperationException(SR.Xml_InvalidOperation);
         }
 
+        public override void WriteValue(decimal value)
+        {
+            Span<char> destinationSpan = stackalloc char[64];
+            if (XmlConvert.TryFormat(value, destinationSpan, out int charsWritten))
+                WriteSpan(destinationSpan.Slice(0, charsWritten));
+            else
+                base.WriteValue(value);
```

Comment:
Nit: should use braces here
```suggestion
            if (XmlConvert.TryFormat(value, destinationSpan, out int charsWritten))
            {
                WriteSpan(destinationSpan.Slice(0, charsWritten));
            }
            else
            {
                base.WriteValue(value);
            }
```